server/services/jwt/jwt.service.js

```diff
@@ -0,0 +1,153 @@
+import jwt from "jsonwebtoken";
+import { ACCESS_EXP_IN, JWT_PWD, REFRESH_EXP_IN } from "./jwt.config.js";
+import { userTokenModel } from "./jwttoken.model.js";
+
+/**
+ * @description Method to create JWT and refresh tokens
+ * @param {*} payload
+ * @returns {Object} Contains the generated access and refresh tokens
+ */
+export const generateToken = async (payload) => {
+  try {
+    if (!JWT_PWD || !payload.userId || !ACCESS_EXP_IN || !REFRESH_EXP_IN) {
+      throw new Error("Missing required environment variables.");
+    }
+
+    // Generate access token
+    const accessToken = jwt.sign(payload, JWT_PWD, {
+      expiresIn: ACCESS_EXP_IN,
+    });
+
+    // Generate refresh token with 1 year validity
+    const refreshToken = jwt.sign(payload, JWT_PWD, {
+      expiresIn: REFRESH_EXP_IN,
+    });
+
+    // Find user token data in the database
+    const userTokenData = await userTokenModel.findOne({
+      userId: payload.userId,
+    });
+
+    if (userTokenData) {
```

Comment:
@Manas0410 - Can we use guard block approach here?

```
if(!userTokenData){
 // handle no token response
 return
}

// Rest code
```

Reply:
Fixed